mk/disable/doc-gen.mk

```diff
@@ -0,0 +1,8 @@
+# This file is only active for `./configure --disable-doc-gen`.
```

Comment:
I don't get what all these `mk/disable` files are for. Surely that can just be replaced by an `ifeq ($(FEATURE), yes) ...` in the appropriate files?

Reply:
IIUC they will print a proper message if you still try to use those make targets when the feature is disabled.

Reply:
They are pulled out the source of unused components can be excluded of the build (this was already the case with the prior `mk/disable-tests.mk`) but yes there is other ways to do this, like just putting them in `Makefile` or whatever guarded by the `if..else..endif`.

As @fricklerhandwerk says, yes the only reason to have them is a nicer error message.

Reply:
My point is: why does the "disable" stuff need to be in a separate file? That seems unnecessary clutter and makes it harder to figure out what's going on. Just write
```
ifeq ($(...), yes)
...
else
target:
    echo "this is disabled"
fi
```
I.e. exactly what we were already doing in `doc/internal-api/local.mk`.

Reply:
It makes it so that the main file can be filtered out, helping to avoid rebuilds a bit, and keeping the same complexity out of the source filter.